docs/components/table-view.md

```diff
@@ -0,0 +1,1011 @@
+  target?: string
+}
+
+type TableData = Array<TableDataEntry>
+```
+
+<KTableView
+  :data="basicDataSortable"
+  :headers="basicHeaders(true, 'username', 'email')"
+  @sort="sortBasicData"
+>
+  <template #actions-items>
+    <KDropdownItem>
+      Edit
+    </KDropdownItem>
+    <KDropdownItem
+      danger
+      has-divider
+    >
+      Delete
+    </KDropdownItem>
+  </template>
+</KTableView>
+
+```vue
+<template>
+  <KTableView
+    :data="tableData"
+    :headers="headers"
+    @sort="sortData"
+  >
+    <template #actions-items>
+      <KDropdownItem>
+        Edit
+      </KDropdownItem>
+      <KDropdownItem
+        danger
+        has-divider
+      >
+        Delete
+      </KDropdownItem>
+    </template>
+  </KTableView>
+</template>
+
+<script setup lang="ts">
+import type { TableViewHeader, TableData, TableDataEntry } from '@kong/kongponents'
+
+const headers: Array<TableViewHeader> = [
+  {
+    key: 'name', 
+    label: 'Full Name', 
+  },
+  {
+    key: 'username',
+    label: 'Username',
+    sortable: true,
+    tooltip: 'Unique for each user.',
+  },
+  {
+    key: 'email',
+    label: 'Email',
+    hidable: true,
+  },
+  {
+    key: 'actions',
+    label: 'Row actions',
+  },
+]
+
+const tableData = ref<TableData>([
+  {
+    id: 1,
+    // notice that property keys in data object correspond to each respective key in headers const
+    name: 'Leanne Graham', 
+    username: 'Bret',
+    email: 'Sincere@april.biz'
+  },
+  {
+    id: 2,
+    name: 'Ervin Howell',
+    username: 'Antonette',
+    email: 'Shanna@melissa.tv'
+  },
+  ...
+])
+
+const sortData = (sortData: TableSortPayload): void => {
+  const data = [...tableData.value]
+  const { sortColumnKey, sortColumnOrder } = sortData || { sortColumnKey: 'username', sortColumnOrder: 'asc' }
+
+  data.sort((a: TableDataEntry, b: TableDataEntry) => {
+    if (sortColumnKey === 'username') {
+      if (sortColumnOrder === 'asc') {
+        if (a.username > b.username) {
+          return 1
+        } else if (a.username < b.username) {
+          return -1
+        }
+
+        return 0
+      } else {
+        if (a.username > b.username) {
+          return -1
+        } else if (a.username < b.username) {
+          return 1
+        }
+
+        return 0
+      }
+    }
+
+    return 0
+  })
+
+  tableData.value = data
+}
+</script>
+```
+
+:::tip NOTE
+Notice that in the example above the _Username_ column is `sortable` and the _Email_ column is `hidable`.
+:::
+
+#### Reserved Data Keys
+
+:::warning NOTE
+This feature is only available in KTableView and not available in [KTable](/components/table).
+:::
+
+- `to` - if you wish to make the whole row a link, you can pass `to` parameter in each `data` entry. If an object is passed, it will be rendered as `<router-link>` component or otherwise if a string is passed, it will be rendered as an anchor tag. Clicking anywhere on table row will trigger route change. Refer to the example below for usage.
```

Comment:
> this would require mutating table data

Since data manipulations are handled by the host app rather than KTableView component, I think this is acceptable?

Right-click support is a deal breaker and a long awaited feature for table component. In my opinion, a reserved key is the easiest way to turn the entire row into a link. I'm not sure how we could implement that using a slot like you suggested. Could you provide more details as to how you imagine I could make it work?

Reply:
> Since data manipulations are handled by the host app rather than KTableView component, I think this is acceptable?

A component that is meant to display data shouldn’t restrict the data/column names for that data. 

I’m not sure anything I mentioned involves right-click support? The table row itself can be a link, but it shouldn’t be tied to a property of the data object the table is displaying. 

It could be a config on the cell itself, slot content, or maybe even a factory function for each row to generate the link.

Reply:
How about a new prop `rowLink: Function`. It would take a function as a parameter and the reason for that is because we need a way for component to pass back row value so that the function returns the correct dynamic route.

Reply:
Actually I think this is something we can introduce and support in both KTable and KTableView (I think we should maintain parity between the two components, to an extend).

Reply:
So I decided to use `rowAttrs` prop for this. The prop takes a function that returns an object which have optional parameters `to` and `target` for making the whole row a link (see [docs](https://deploy-preview-2338--kongponents.netlify.app/components/table-view.html#row-links)). I think this is much better than including those keys in `data` prop. Lmk what you think.

Reply:
Utilizing `rowAttrs` here is a bit confusing to be honest. Turning a row into a link isn't really related to HTML attributes being passed to the row (e.g. like `data-testid`, etc.)

It's better than the previous iteration, but is there a reason there can't be a separate prop specifically for a row link?

Reply:
I think it can be - I was just trying to avoid adding an extra arbitrary prop but maybe it's better to just add one rather than complicate existing ones.. is that your thinking here too?

Reply:
Yes I think since having "row links" is a net-new feature, adding it as a separate row-level "config" is ideal. So it can work similar to `rowAttrs` but something like `rowLink` instead